packages/design-system/src/components/TextField/useLabelMask.test.tsx

```diff
@@ -1,8 +1,116 @@
-import useLabelMask, { DATE_MASK } from './useLabelMask';
+import useLabelMask, {
+  DATE_MASK,
+  SSN_MASK,
+  PHONE_MASK,
+  ZIP_MASK,
+  CURRENCY_MASK,
+  MaskFunction,
+} from './useLabelMask';
 import { render, screen } from '@testing-library/react';
 import { renderHook } from '@testing-library/react-hooks';
 import userEvent from '@testing-library/user-event';
 
+describe('SSN_MASK', () => {
+  it('returns just the mask when given no input', () => {
+    expect(SSN_MASK('')).toEqual('XXX-XX-XXXX');
+  });
+
+  it('masks complete social security numbers', () => {
+    expect(SSN_MASK('123-45-6789')).toEqual('123-45-6789');
+    expect(SSN_MASK('123 45 6789')).toEqual('123-45-6789');
+    expect(SSN_MASK('123456789')).toEqual('123-45-6789');
+    expect(SSN_MASK('123.45.6789')).toEqual('123-45-6789');
+  });
+
+  it('masks incomplete social security numbers', () => {
+    expect(SSN_MASK('123')).toEqual('123-XX-XXXX');
+    expect(SSN_MASK('1234')).toEqual('123-4X-XXXX');
+    expect(SSN_MASK('1234567')).toEqual('123-45-67XX');
+  });
+
+  it('handles valueOnly parameter', () => {
+    expect(SSN_MASK('123', true)).toEqual('123');
+    expect(SSN_MASK('12345', true)).toEqual('123-45');
+    expect(SSN_MASK('1234567', true)).toEqual('123-45-67');
+  });
+});
+
+describe('ZIP_MASK', () => {
+  it('returns just the mask when given no input', () => {
+    expect(ZIP_MASK('')).toEqual('XXXXX');
+  });
+
+  it('masks complete zip codes', () => {
+    expect(ZIP_MASK('12345')).toEqual('12345');
+  });
+
+  it('masks incomplete social zip codes', () => {
+    expect(ZIP_MASK('1')).toEqual('1XXXX');
+    expect(ZIP_MASK('12')).toEqual('12XXX');
+    expect(ZIP_MASK('123')).toEqual('123XX');
+    expect(ZIP_MASK('1234')).toEqual('1234X');
+  });
+
+  it('handles valueOnly parameter', () => {
+    expect(ZIP_MASK('1', true)).toEqual('1');
+    expect(ZIP_MASK('12', true)).toEqual('12');
+    expect(ZIP_MASK('123', true)).toEqual('123');
+  });
+});
+
+describe('CURRENCY_MASK', () => {
+  it('returns just the mask when given no input', () => {
+    expect(CURRENCY_MASK('')).toEqual('$0.00');
+  });
+
+  it('returns correctly formatted dollar amounts', () => {
+    expect(CURRENCY_MASK('1')).toEqual('$1.00');
+    expect(CURRENCY_MASK('12.479')).toEqual('$12.48');
```

Comment:
Just curious - do we expect the `.479` here to round up, or should we trim any chars after 2 decimal places? `.48` vs `.47`?

I also think this is the plot to Office Space.

Reply:
I would probably vote for trimming. What does the old one do?